### ISSUE TITLE
Can't start docker instance. Spinning circle.

### ISSUE THREAD

**Issue by alekc** [theia-ide/theia-apps#92]
Tried with both :latest and :next

```
docker run -it -p 3000:3000 -v "$(pwd):/home/project:cached" theiaide/theia:latest
Unable to find image 'theiaide/theia:latest' locally
latest: Pulling from theiaide/theia
4fe2ade4980c: Already exists
eeb7d76f44e7: Already exists
e35f88fcc259: Already exists
6efbb7e943db: Pull complete
ebc56766b478: Pull complete
46f9949243cd: Pull complete
99718621530c: Pull complete
Digest: sha256:b8cb945cd5496ec19beb88ea54d9435efae55be9d756534e938143357fd724ed
Status: Downloaded newer image for theiaide/theia:latest
yarn run v1.9.4
$ /home/theia/node_modules/.bin/theia start /home/project --hostname=0.0.0.0
Starting server worker...
Server worker has been started. [ID: 1 | PID: 37]
Failed to resolve module: electron
Server worker is ready to accept messages. [ID: 1 | PID: 37]
root INFO Theia app listening on http://0.0.0.0:3000.
Received message which is neither a response nor a notification message:
"3000"
root INFO [nsfw-watcher: 43] Started watching: /home/project
root INFO [nsfw-watcher: 43] Started watching: /home/project
root INFO [nsfw-watcher: 63] Started watching: /home/project
root INFO [nsfw-watcher: 63] Started watching: /home/project
root WARN Collided keybinding is ignored;  {"command":"monaco.editor.action.wordHighlight.next","keybinding":"f7","context":"editorTextFocus"}  collided with  {"command":"monaco.editor.action.diffReview.next","keybinding":"f7","context":"editorTextFocus"}
root WARN Could not register keybinding:
  {"command":"monaco.editor.action.wordHighlight.next","keybinding":"f7","context":"editorTextFocus"}
Error: "f7" is in collision with something else [scope:0]
root WARN Collided keybinding is ignored;  {"command":"monaco.editor.action.wordHighlight.prev","keybinding":"shift+f7","context":"editorTextFocus"}  collided with  {"command":"monaco.editor.action.diffReview.prev","keybinding":"shift+f7","context":"editorTextFocus"}
root WARN Could not register keybinding:
  {"command":"monaco.editor.action.wordHighlight.prev","keybinding":"shift+f7","context":"editorTextFocus"}
Error: "shift+f7" is in collision with something else [scope:0]
root INFO [nsfw-watcher: 87] Started watching: /home/project
root INFO [nsfw-watcher: 87] Started watching: /home/project
root INFO 'USE_LOCAL_GIT' is set to true. Trying to use local Git for 'dugite' execution.
root INFO 'USE_LOCAL_GIT' is set to true. Trying to use local Git for 'dugite' execution.
root INFO 'USE_LOCAL_GIT' is set to true. Trying to use local Git for 'dugite' execution.
root INFO Using external Git executable. Git path: /usr/bin/git. Git exec-path: /usr/libexec/git-core. [Version: 2.18.0]
root INFO Using external Git executable. Git path: /usr/bin/git. Git exec-path: /usr/libexec/git-core. [Version: 2.18.0]
root INFO Using external Git executable. Git path: /usr/bin/git. Git exec-path: /usr/libexec/git-core. [Version: 2.18.0]
root WARN Collided keybinding is ignored;  {"command":"monaco.editor.action.wordHighlight.next","keybinding":"f7","context":"editorTextFocus"}  collided with  {"command":"monaco.editor.action.diffReview.next","keybinding":"f7","context":"editorTextFocus"}
root WARN Could not register keybinding:
  {"command":"monaco.editor.action.wordHighlight.next","keybinding":"f7","context":"editorTextFocus"}
Error: "f7" is in collision with something else [scope:0]
root WARN Collided keybinding is ignored;  {"command":"monaco.editor.action.wordHighlight.prev","keybinding":"shift+f7","context":"editorTextFocus"}  collided with  {"command":"monaco.editor.action.diffReview.prev","keybinding":"shift+f7","context":"editorTextFocus"}
root WARN Could not register keybinding:
  {"command":"monaco.editor.action.wordHighlight.prev","keybinding":"shift+f7","context":"editorTextFocus"}
Error: "shift+f7" is in collision with something else [scope:0]
root INFO [nsfw-watcher: 169] Started watching: /home/project
root INFO [nsfw-watcher: 169] Started watching: /home/project
````

When accessing docker instance (192.168.0.33:3000) all I can see is a spinning circle on the black screen. 
On chrome I get following in console:
```
contribution-provider.ts:67 Uncaught (in promise) TypeError: e is not a function
    at Object.t.bindContributionProvider (contribution-provider.ts:67)
    at p (navigator-container.ts:52)
    at Object.t.createFileNavigatorWidget (navigator-container.ts:58)
    at e.dynamicValue (navigator-frontend-module.ts:38)
    at resolver.js:63
    at c (resolver.js:10)
    at resolver.js:63
    at Object.t.resolve (resolver.js:96)
    at container.js:319
    at e._get (container.js:310)
```

**Comment by marcdumais-work** [comment]
> Probably caused by this bug: theia-ide/theia#3204

The issue above is now fixed in master. So the problem should be gone, for at least the ```next``` versions of our images. 

For imaged based on ```latest``` I expect it should work again after we do a release, tentatively this Thursday. So Let's re-test and hopefully close this issue in a few days.

**Comment by sr229** [comment]
Probably needs to close this now as well 

I should update my copy of Theia images as well since they may carry the broken version

**Comment by marcdumais-work** [comment]
> Probably needs to close this now as well

ATM the "next" versions of the images should work, but the "latest" ones will only get the fix after we release Theia 0.3.16, tentatively tomorrow.

> I should update my copy of Theia images as well since they may carry the broken version

+1

**Comment by legenddcr** [comment]
My Chrome browser exception is:

logger-protocol.ts:99 root WARN Could not register keybinding:
  {"command":"monaco.editor.action.wordHighlight.prev","keybinding":"shift+f7","context":"editorTextFocus"}
Error: "shift+f7" is in collision with something else [scope:0]
e.log @ logger-protocol.ts:99

**Comment by marcdumais-work** [comment]
Hi @legenddcr 

What's the symptom that you see? Does the front-end just display the spinning circle indefinitely? The backend error in the log should not prevent Theia from working.

Just to rule-out a common source of problems: if you had previously used that image, you may need to explicitly pull for the later version, or you'll still be running the older one you already have locally: 
```
docker pull theiaide/theia:next
```
And if it downloaded a new version, then re-try running the image.

**Comment by legenddcr** [comment]
@marcdumais-work Thanks for your help. I can run it locally now :)

**Comment by marcdumais-work** [comment]
@alekc  I just re-tested and the `theiaide/theia:latest` published yesterday evening, based on Theia `0.3.16` now starts correctly for me. Should be the same for the other images, baring new issues.

Closing this issue; re-open if you still see the problem

**Comment by xihaoxiaohao** [comment]
I pulled the next image today and still shows the error:

````
Starting the master backend process with 5000 (ms) timeout.
Starting server worker...
Server worker has been started. [ID: 1 | PID: 11]
Failed to resolve module: electron
Server worker is ready to accept messages. [ID: 1 | PID: 11]
root INFO Theia app listening on http://0.0.0.0:3000.
Received message which is neither a response nor a notification message:
"3000"
root WARN Collided keybinding is ignored;  {"command":"monaco.editor.action.wordHighlight.next","keybinding":"f7","context":"editorTextFocus"}  collided with  {"command":"monaco.editor.action.diffReview.next","keybinding":"f7","context":"editorTextFocus"}
root WARN Could not register keybinding:
  {"command":"monaco.editor.action.wordHighlight.next","keybinding":"f7","context":"editorTextFocus"}
Error: "f7" is in collision with something else [scope:0]
root WARN Collided keybinding is ignored;  {"command":"monaco.editor.action.wordHighlight.prev","keybinding":"shift+f7","context":"editorTextFocus"}  collided with  {"command":"monaco.editor.action.diffReview.prev","keybinding":"shift+f7","context":"editorTextFocus"}
root WARN Could not register keybinding:
  {"command":"monaco.editor.action.wordHighlight.prev","keybinding":"shift+f7","context":"editorTextFocus"}
Error: "shift+f7" is in collision with something else [scope:0]
````
